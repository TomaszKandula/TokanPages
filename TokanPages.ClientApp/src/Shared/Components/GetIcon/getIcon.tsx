import * as React from "react";
import { 
    Apple, 
    Assignment, 
    Build, 
    ContactMail, 
    DirectionsBike, 
    Gavel, 
    Home, 
    MusicNote, 
    Person, 
    PersonAdd, 
    PhotoCamera, 
    Policy, 
    Star, 
    Subject,
    SportsSoccer, 
    ViewList, 
    VpnKey,
    Lock,
    Code,
    GitHub,
    MenuBook,
    Edit,
} from "@material-ui/icons";

interface IProperty
{
    iconName: string;
}

export const GetIcon = (props: IProperty): JSX.Element =>
{
    switch(props.iconName)
    {
        case "Person": return <Person />;
        case "PersonAdd": return <PersonAdd />;
        case "Home": return <Home />;
        case "ViewList": return <ViewList />;
        case "Subject": return <Subject />;
        case "Build": return <Build />;
        case "Assignment": return <Assignment />;
        case "Star": return <Star />;
        case "PhotoCamera": return <PhotoCamera />;
        case "SportsSoccer": return <SportsSoccer />;
        case "MusicNote": return <MusicNote />;
        case "DirectionsBike": return <DirectionsBike />;
        case "ContactMail": return <ContactMail />;
        case "Gavel": return <Gavel />;
        case "Policy": return <Policy />;
        case "VpnKey": return <VpnKey />;
        case "Lock": return <Lock />; 
        case "Code": return <Code />; 
        case "GitHub": return <GitHub />; 
        case "MenuBook": return <MenuBook />; 
        case "Edit": return <Edit />; 

        default: return <Apple />;
    }
}
