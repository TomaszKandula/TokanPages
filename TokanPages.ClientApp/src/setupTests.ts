// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
