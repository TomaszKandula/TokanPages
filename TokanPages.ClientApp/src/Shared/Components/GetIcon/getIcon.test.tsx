import "../../../setupTests";
import React from "react";
import { shallow } from "enzyme";
import { GetIcon } from "./getIcon";

describe("test render function 'getIcon'", () => {
    const NameList: string[] = [
        "Person",
        "PersonAdd",
        "Home",
        "ViewList",
        "Subject",
        "Build",
        "Assignment",
        "Star",
        "PhotoCamera",
        "SportsSoccer",
        "MusicNote",
        "DirectionsBike",
        "ContactMail",
        "Gavel",
        "Policy",
        "VpnKey",
        "Lock",
        "Unknown element renders apple icon",
    ];

    const wrapper = shallow(
        <div>
            {NameList.map((item, index) => (
                <GetIcon key={index} iconName={item} />
            ))}
        </div>
    );

    it("should render SVG icon based on input string.", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
