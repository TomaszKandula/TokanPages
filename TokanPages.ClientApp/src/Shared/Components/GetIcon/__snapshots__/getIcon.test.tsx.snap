// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render function 'getIcon' should render SVG icon based on input string. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          person
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          personadd
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          home
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          viewlist
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          subject
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          build
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          assignment
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          star
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          photocamera
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          sportssoccer
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          musicnote
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          directionsbike
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          contactmail
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          gavel
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          policy
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          vpnkey
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          lock
        </span>
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root"
        >
          unknown element renders apple icon
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        person
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        personadd
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        home
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        viewlist
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        subject
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        build
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        assignment
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        star
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        photocamera
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        sportssoccer
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        musicnote
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        directionsbike
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        contactmail
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        gavel
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        policy
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        vpnkey
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        lock
      </span>
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root"
      >
        unknown element renders apple icon
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
